lib/services/local/access_list.go

```diff
@@ -223,5 +224,5 @@ func (a *AccessListService) DeleteAllAccessListMembers(ctx context.Context) erro
 }
 
 func lockName(accessListName string) string {
-	return string(backend.Key("access_list", accessListName))
+	return path.Join("access_list", accessListName)
```

Comment:
Looks like this is the "key name function" used for other locks https://github.com/gravitational/teleport/blob/master/lib/backend/backend.go#L423
I'm not sure if this PR is a good place, but I think we should unify the logic somehow, to prevent issues like that in the future.

Reply:
I think most use cases aren't subscoped locks like this, they're more applicable to the whole service.

A big step in catching these is figuring out why the key isn't being Sanitized during this test, though. This PR is necessary at present though, as I think it inadvertently broke `e` tests.

Reply:
I just updated this to use the `Sanitizer` in the tests, which triggers the problem with the original `lockName` implementation.